src/evagram/website/backend/api/models.py

```diff
@@ -29,33 +50,52 @@ class Meta:
         unique_together = (('variable_name', 'channel'),)
 
 
-class Observations(models.Model):
-    observation_id = models.AutoField(primary_key=True)
-    observation_name = models.CharField(null=False, unique=True, default="null")
+class Groups(models.Model):
+    group_id = models.AutoField(primary_key=True)
+    group_name = models.CharField(unique=True, null=False, default="null")
 
     class Meta:
-        db_table = 'observations'
+        db_table = 'groups'
 
 
-class Owners(models.Model):
-    owner_id = models.AutoField(primary_key=True)
-    first_name = models.CharField(null=True)
-    last_name = models.CharField(null=True)
-    username = models.CharField(null=False, unique=True, default="null")
-
-    class Meta:
-        db_table = 'owners'
+class PlotType(models.TextChoices):
+    SCATTER = ("scatter", "Scatter Plot")
+    MAP_SCATTER = ("map_scatter", "Map Scatter Plot")
+    MAP_GRIDDED = ("map_gridded", "Map Gridded")
+    DENSITY = ("density", "Density Graph")
+    HISTOGRAM = ("histogram", "Histogram")
+    LINE = ("line", "Line Graph")
 
 
 class Plots(models.Model):
     plot_id = models.AutoField(primary_key=True)
+    plot_type = models.CharField(choices=PlotType.choices, null=True)
     div = models.CharField(blank=True, null=True)
     script = models.CharField(blank=True, null=True)
-    experiment = models.ForeignKey(Experiments, models.CASCADE)
-    group = models.ForeignKey(Groups, models.CASCADE)
-    observation = models.ForeignKey(Observations, models.CASCADE)
-    variable = models.ForeignKey(Variables, models.CASCADE)
+    begin_cycle_time = models.DateTimeField(null=False)
```

Comment:
For a `DateTimeField`, Django stores these fields in the model in UTC with an offset depending on what timezone the data comes from (https://docs.djangoproject.com/en/dev/topics/i18n/timezones/?ref=hunj.dev#time-zones). In `evagram_input`, `begin_cycle_time` should be processed as UTC because it is labeled with a 'Z' at the end which indicates zero hour offset or UTC time. For it to be saved in the database with an offset of -4/-5 means somewhere it's reading in the cycle times as EDT/EST. I think that discrepancy is in `evagram_input` here: https://github.com/GEOS-ESM/evagram_input/blob/e85c0d689afc28158395cfbf05311a5cfacbd3e6/src/evagram_input/input_tool/input_tool.py#L120. In the templates, i.e. `%Y%m%dT%H%M%SZ`, the term `Z` is read in as a constant string value and not an indicator that it's UTC time. This results in a `datetime` object with no timezone information. Then, this datetime object is inserted into the database with no further processing. I wonder if the `None` timezone value is defaulting to the machine's timezone. Adding explicit time zone information to `return datetime.strptime(cycle_time_str, fmt)` should fix this.

Reply:
Given that we are defaulting to UTC values for the entire system, we shouldn't expect to see timezone offsets anywhere.

Reply:
Thank you for taking the time to read through all my code and finding where the discrepancy is! I didn't realize that the label 'Z' had any significance, but it does indicate UTC time. I looked through the `datetime` docs and found that `%z` does indeed parse the Z in the datetime string and adds the utc timezone context into the datetime object. I will update `evagram_input` on my end and see if it fixes the offset.

Reply:
I think `%z` might be for handling numerical timezone offsets (such as 20240807T000000 **+5500**). To add timezone info to a datetime object, try something like `datetime_object.astimezone(datetime.timezone.utc)`

Reply:
Got it, thanks!